src/engine/cachingreader/cachingreaderworker.cpp

```diff
@@ -125,16 +131,23 @@ void CachingReaderWorker::loadTrack(const TrackPointer& pTrack) {
     // Unload the track
     m_pAudioSource.reset(); // Close open file handles
 
-    if (!pTrack) {
-        // If no new track is available then we are done
-        const auto update = ReaderStatusUpdate::trackUnloaded();
-        m_pReaderStatusFIFO->writeBlocking(&update, 1);
-        return;
-    }
+    const auto update = ReaderStatusUpdate::trackUnloaded();
+    m_pReaderStatusFIFO->writeBlocking(&update, 1);
+}
 
+void CachingReaderWorker::loadTrack(const TrackPointer& pTrack) {
     // Emit that a new track is loading, stops the current track
     emit trackLoading();
 
+    m_pAudioSource.reset(); // Close open file handles of the old track.
```

Comment:
This code is duplicated in ejectTrack() with a slightly different ordering. I don't understand why.

Reply:
There is no reason. I have introduced cleanUpOldTrack() to make it clear.